PigLatin/PigLatin.cs

```diff
@@ -7,15 +7,125 @@ class Program
+            string VowelNewWord = TranslateWordVowel(userInput);
+            string startsWithVowelNewWord = startsWithVowel(userInput);
+            string lowercaseNewWord = makeLowerCase(userInput);
+            Console.Write("heres your word translated to piglatin:     ");
+            Console.WriteLine(newWord);
 
+            Console.Write("heres your word translated at the first vowel to piglatin:     ");
+            Console.WriteLine(VowelNewWord);
+
+            Console.Write("heres your word translated to piglatin unless it starts with a vowel:     ");
+            Console.WriteLine(startsWithVowelNewWord);
+
+            Console.Write("heres your word translated to piglatin and made lowercase:     ");
+            Console.WriteLine(lowercaseNewWord);
+            Console.WriteLine("");
+
+            Console.WriteLine("Let's try a whole sentence!");
+            string userInputSentence = Console.ReadLine();
+            string translatedSentence = moreWords(userInputSentence);
+            Console.WriteLine(translatedSentence);
+            Console.Write("oops let me fix that punctuation");
+            for (int i = 0; i < 5; i++)
+            {
+                System.Threading.Thread.Sleep(1000);
+                Console.Write(".");
+            }
+            Console.WriteLine("");
+            string fixTranslation = fixPunctuation(translatedSentence);
+            Console.WriteLine(fixTranslation);
             // leave this command at the end so your program does not close automatically
             Console.ReadLine();
         }
-        
+
         public static string TranslateWord(string word)
+        {
+            string firstLetter = word.Substring(0, 1);
+            string restWord = word.Substring(1);
+            string translatedWord = (restWord + firstLetter + "ay");
+            return translatedWord;
+        }
+
+        public static string TranslateWordVowel(string word)
         {
             // your code goes here
-            return word;
+            int firstVowelIndex = word.IndexOfAny(new char[] { 'a', 'e', 'i', 'o', 'u', 'y' });
+            string firstPart = word.Substring(0, firstVowelIndex);
+            string secondPart = word.Substring(firstVowelIndex);
+            string translatedWord = (secondPart + firstPart + "ay");
+
+            return translatedWord;
+        }
+
+        public static string startsWithVowel(string word)
+        {
+            // your code goes here
+            int firstVowelIndex = word.IndexOfAny(new char[] { 'a', 'e', 'i', 'o', 'u', 'y' });
+            string last = word.Substring(0, 1);
+            string first = word.Substring(1);
+            if (firstVowelIndex == 0)
+            {
+
+                string translatedWord = (last + first + "yay");
+
+                return translatedWord;
+            }
+            else
+            {
+                string notAVowel = (first + last + "ay (try a word that starts with a vowel!)");
```

Comment:
While not in the assignment, another case you could have checked for is words that end in y and start with a vowel. In cases such as those, "agency" for example, you end up with a double "y" (e.g. agencyyay).

Reply:
Ill build that in.